### ISSUE TITLE
proper way to further train a model

### ISSUE THREAD

**Issue by tino926** [ultralytics/yolov5#7951]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I need a suggestion to further train an existing model.

I already trained a model with 1200 epochs. From the mAP curve, I expect better result can be obtained with more epochs.
![image](https://user-images.githubusercontent.com/2639399/169976470-ddd5cb0b-0b30-482e-a240-1ca12dd80c81.png)

Since I already have a trained model, I want to train from it. Thus I use it as the initial to start a new training process with smaller learning rate (blue lines). 
![image](https://user-images.githubusercontent.com/2639399/169995822-9c162454-dd51-4c7b-ae71-46302b82b960.png)
Unexpectedly, the optimization process is very slow: even after 200 epochs, the training loss (blue curves) still can't catch the final training loss (purple curves) from the previous run.
![image](https://user-images.githubusercontent.com/2639399/169978561-98cff557-f039-4aa9-a52b-befa9e2575ef.png)

For the first training process, I used the hyp.scratch-low.yaml.
For the second, I only modified lr0, warmup_epochs, and warmup_bias_lr: 
![image](https://user-images.githubusercontent.com/2639399/169980471-d9292012-eaff-4f91-8747-459aa66447aa.png)
and the learning rate evolution is
![image](https://user-images.githubusercontent.com/2639399/169995822-9c162454-dd51-4c7b-ae71-46302b82b960.png)

with this command:
```bash
CUDA_VISIBLE_DEVICES=1 python \
       train.py --project ./runs/r5_yv5ss6_480_coco6 \
       --cfg ./models/yolov5ss6_conv_sppf_c3n.yaml \
       --hyp ./data/hyps/hyp.scratch-low_2.yaml \
       --weights ./runs/r5_yv5ss6_480_coco6/bkgsam_ss_conv_sppf_c3n_scratch_1200/weights/best.pt \
       --patience 600 \
       --img 480 --multi-scale --epoch 1200 \
       --name bkgsam_ss_conv_sppf_c3n_scratch_2_1200 \
       --data ./data/coco6_bkg_sample_ramdisk_v.yaml \
       --batch-size 128 \
       --workers 8 
```

I want to ask:
1. Did I miss something in the hyper parameter file or in the training command so that the training losses are not as low as those in previous run?
2. If I want to train a model trained with more epochs, is the best practice just to forget the first run and train it again (with the same hyper parameter file and a larger epoch number)?

**Comment by glenn-jocher** [comment]
@tino926 it's expected that higher LR will produce higher loss, therefore your blue curve is behaving as expected since it has a higher LR than the final pink LR.

Finetuning generally requires a different set of hyps, i.e. VOC hyps in hyps dir are evolved for 300 generations on a finetuning base scenario starting from COCO pretrained weights:
https://github.com/ultralytics/yolov5/blob/a3a652c9334a7e7e97dc73d83df31f06bbac49c8/data/hyps/hyp.VOC.yaml#L2-L4

This requires significant compute expense obviously but is the best approach.

**Comment by haimat** [comment]
> @tino926 it's expected that higher LR will produce higher loss, therefore your blue curve is behaving as expected since it has a higher LR than the final pink LR.
> 
> Finetuning generally requires a different set of hyps, i.e. VOC hyps in hyps dir are evolved for 300 generations on a finetuning base scenario starting from COCO pretrained weights:
> 
> https://github.com/ultralytics/yolov5/blob/a3a652c9334a7e7e97dc73d83df31f06bbac49c8/data/hyps/hyp.VOC.yaml#L2-L4
> 
> This requires significant compute expense obviously but is the best approach.

Hello @glenn-jocher, could you please be so kind and elaborate on this? The OP said that he even decreased the `lr0` value in the second training, so why does this still end up in a higher LR then the final pink LR?

**Comment by glenn-jocher** [comment]
@haimat 👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/aa542ce6a65658ff931fee9bbab77c0145c152f0/train.py#L175-L181) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="Screenshot 2022-04-10 at 11 11 51" src="https://user-images.githubusercontent.com/26833433/162611202-4e0b611f-6cca-4e43-94fc-bbc44e6447f8.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.run --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.run --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck 🍀 and let us know if you have any other questions!
